.github/workflows/integration-test.yaml

```diff
@@ -150,22 +157,22 @@ jobs:
 
       - name: Password protect Playwright traces
         id: password_protect_test_results
-        if: ${{ always() && inputs.run-playwright }}
+        if: ${{ always() && matrix.test == 'playwright' && inputs.run-playwright }}
         shell: bash
         env:
           ZIP_PASSWORD: ${{ secrets.TEST_USER_PASSWORD }}
         run: 7z a ./playwright-traces.7z -mx=0 -mmt=off ./frontend/test-results -p"$ZIP_PASSWORD"
       - name: Publish unit test results
         uses: EnricoMi/publish-unit-test-result-action/composite@8885e273a4343cd7b48eaa72428dea0c3067ea98 # v2.14.0
-        if: ${{ always() && !env.act }}
+        if: ${{ always() && matrix.test == 'dotnet' && !env.act }}
         with:
           check_name: Integration Tests ${{ inputs.runs-on }} for Mercurial ${{ inputs.hg-version }}
           files: ./test-results/*.trx
       - name: Publish results to testspace
-        if: always()
+        if: ${{ always() && matrix.test == 'dotnet' }}
```

Comment:
this step is uploading playwright results too, so it needs to run in both jobs

Reply:
Good to know; when I revert my earlier commit and switch to using two jobs, I'll make sure this runs after both. (In a third job that depends on the other two).

Reply:
I wouldn't bother as that will be complicated to setup, just duplicate this step in each job.

Reply:
Yep, just realized that myself as I came back to this. Done in commit 1d0e2fe9.